pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/participant/OsteoParticipantService.java

```diff
@@ -118,50 +117,71 @@ private static void createOsteoTag(ElasticSearchParticipantDto participantDto, S
                                        DDPInstance osteoInstance, String tagName) {
         // invariant: participant should have a DSM entity
         Dsm dsm = participantDto.getDsm().orElseThrow();
+        createOteoTag(dsm, ddpParticipantId, osteoInstance, tagName);
+    }
+
+    private static void createOteoTag(Dsm dsm, String ddpParticipantId, DDPInstance osteoInstance, String tagName) {
+        log.info("Creating {} cohort tag for participant {}", tagName, ddpParticipantId);
+        CohortTag cohortTag = new CohortTag(tagName, ddpParticipantId, osteoInstance.getDdpInstanceIdAsInt());
+        int newCohortTagId = cohortTagDao.create(cohortTag);
+        cohortTag.setCohortTagId(newCohortTagId);
 
-        CohortTag newCohortTag =
-                new CohortTag(tagName, ddpParticipantId, osteoInstance.getDdpInstanceIdAsInt());
-        int newCohortTagId = cohortTagDao.create(newCohortTag);
-        newCohortTag.setCohortTagId(newCohortTagId);
+        List<CohortTag> cohortTags = dsm.getCohortTag();
```

Comment:
Could we change `Dsm` so that the method is plural `getCohortTags()` instead of singular, and initialize `cohortTag` to a mutable empty list where the field is declared?  The antipattern of lazy initialization to an immutable empty list on the getter makes for some weird code here.

Reply:
Sure, I can take care of this.
But, in general, there are *so many* collateral improvements I can make in any given PR, that I tend to avoid them, both as a service to the reviewer (less to review), and to leave time for the long backlog of Pepper tasks.

Reply:
Changing the method name caused broader problems, since this is effectively an override of the Lombok getter. I chose to not run down that rabbit hole. Let me know if you feel strongly about the name change and I will deal with those issues.

Reply:
Understood.  Finding the right balance is hard...I was looking for a quick incremental win but perhaps this is more of a distraction.

Reply:
Oh geez, so sorry for another case of "this one easy thing" turning into a sinkhole!  Let's leave it as is.